app/main.py

```diff
@@ -1,3 +1,59 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __add__(self, other: Distance | int | float) -> Distance:
+        if (isinstance(other, Distance)):
+            return Distance(self.km + other.km)
+        return Distance(self.km + other)
+
+    def __iadd__(self, other: Distance | int | float) -> Distance:
+        if (isinstance(other, Distance)):
+            self.km = self.km + other.km
+        else:
+            self.km = self.km + other
+        return self
+
+    def __mul__(self, other: Distance | int | float) -> Distance:
```

Comment:
You annotate that other might be Distance, but don't handle such occasion in code. I think it might be better or to remove Distance from type annotation or handle such case in code like raise some Error. And with  __truediv__ the same